agent/consul/server_connect.go

```diff
@@ -219,21 +218,22 @@ func (s *Server) SignCertificate(csr *x509.CertificateRequest, spiffeID connect.
 
 	// Append any intermediates needed by this root.
 	for _, p := range caRoot.IntermediateCerts {
-		pem = strings.TrimSpace(pem) + "\n" + p
+		pem = ca.AddSingleNewline(pem) + p
```

Comment:
I think we need to ensure a newline on `p` as well here, since we aren't doing it on the vault provider.